src/components/Avatar/AvatarGroup/AvatarGroup.tsx

```diff
@@ -0,0 +1,82 @@
+import { ComponentElement, ComponentState } from "react"
+import { Avatar, AvatarProps } from ".."
+import { AvatarBase } from "../AvatarBase"
+import styles from "./AvatarGroup.module.css"
+
+type AvatarGroupChildren =
+  | ComponentElement<AvatarProps, ComponentState>
+  | ComponentElement<AvatarProps, ComponentState>[]
+
+export interface AvatarGroupProps {
+  max?: number
+  size?: AvatarProps["size"]
+  renderHiddenCount?: (hiddenCount: number) => JSX.Element
+  children: AvatarGroupChildren
+}
+
+export function AvatarGroup({
+  size = "sm",
+  max = 3,
+  children,
+  renderHiddenCount,
+}: AvatarGroupProps) {
+  const childrenArray = Array.isArray(children) ? children : [children]
+  const slicedChildren = childrenArray.slice(0, max)
```

Comment:
I wonder if `visibleChildren` or something more descriptive would be more helpful in understanding what we intend to achieve with slice.